test/core/util/test_config.cc

```diff
@@ -90,11 +92,32 @@ gpr_timespec grpc_timeout_milliseconds_to_deadline(int64_t time_ms) {
           GPR_TIMESPAN));
 }
 
-void grpc_test_init(int /*argc*/, char** argv) {
+namespace {
+void grpc_test_init_parse_argv(int* argc, char** argv) {
+  // flags to look for and consume
+  constexpr char poller_flag[] = "--poller=";
+  constexpr size_t poller_flag_len = sizeof(poller_flag) - 1;
+  for (int i = 1; argc && i < *argc; ++i) {
+    if (strncmp(poller_flag, argv[i], poller_flag_len) == 0) {
+      gpr_setenv("GRPC_POLL_STRATEGY", argv[i] + poller_flag_len);
+      // remove the spent argv
+      for (int j = i; j < *argc - 1; ++j) {
+        argv[j] = argv[j + 1];
+      }
+      --i;
+      --(*argc);
+    }
+  }
+}
+}  // namespace
+
+void grpc_test_init(int* argc, char** argv) {
```

Comment:
Why do we need to pass `argc` in as a pointer?  Do we have any callers that look at this value after we're done with it?

Reply:
Yes, exactly. test/core/fling/server.cc and all of the bm tests that rely on `third_party/benchmark` for example. The global test init parses and consumes a subset of flags, and the rest need to remain for test-specific logic. For the same reason, we can't use absl::flags here.